PiwikTracker/Event.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Event.swift
+//  PiwikTracker
+//
+//  Created by Cornelius Horstmann on 21.12.16.
+//  Copyright © 2016 PIWIK. All rights reserved.
+//
+
+import Foundation
+
+/// Represents an event of any kind.
+///
+/// - Note: Should we store the resolution in the Event (cleaner) or add it before transmission (smaller)?
+/// - Todo: 
+///     - Add Campaign Parameters: _rcn, _rck
+///     - Add Action info
+///     - Event Tracking info
+///     - Add Content Tracking info
+///     - Add Ecommerce info
+///
+/// # Key Mapping:
+/// Most properties represent a key defined at: [Tracking HTTP API](https://developer.piwik.org/api-reference/tracking-api). Keys that are not supported for now are:
+///
+/// - idsite, rec, rand, apiv, res, cookie,
+/// - All Plugins: fla, java, dir, qt, realp, pdf, wma, gears, ag
+/// - cid: We will use the uid instead of the cid.
+/// - dimension: For now we don't support dimension.
+/// - UserAgent (ua) should be set in the dispatcher, right?
+internal struct Event {
+    let uuid: NSUUID
+    let visitor: Visitor
+    let session: Session
+    
+    /// The Date and Time the event occurred.
+    /// api-key: h, m, s
+    let date: Date
+    
+    /// The full URL for the current action. 
+    /// api-key: url
+    let url: URL
+    
+    /// api-key: action_name
+    let actionName: [String]
+    
+    /// The language of the device.
+    /// Should be in the format of the Accept-Language HTTP header field.
+    /// api-key: lang
+    let language: String
+    
+    /// Should be set to true for the first event of a session.
+    /// api-key: new_visit
+    let isNewSession: Bool
+    
+    /// Currently only used for Campaigns
+    /// api-key: urlref
+    let referer: URL?
+    
+    /// api-key: _cvar
+    //let customVariables: [CustomVariable]
+}
+
+internal struct Visitor {
+    /// Unique ID per visitor (device in this case). Should be
+    /// generated upon first start and never changed after.
+    /// api-key: _id
+    let id: String
+    
+    /// An optional user identifier such as email or username.
+    /// api-key: uid
+    let userId: String?
+}
+
+internal struct Session {
+    /// The number of sessions of the current user.
+    /// api-key: _idvc
+    let sessionsCount: Int
+    
+    /// The timestamp of the previous visit.
+    /// Discussion: Should this be now for the first request?
```

Comment:
> What should the values for _viewts and _idts be for the first session/visit/request? Is there any reason to not set it to the current date and time?

@brototyp It is fine to set _viewts and _idts on the first visit as well. Piwik Tracking API will use these timestamps to process a number of "days since", which will be zero, so it'll work as expected.